cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -1295,6 +1295,75 @@ public void testNoKafkaCluster(VertxTestContext context) {
                 })));
     }
 
+    /**
+     * Tests that `KafkaRebalance` stays in `Ready` state when the Kafka cluster moves to `NotReady` state.
+     *
+     * 1. A new KafkaRebalance resource is created. It is moved directly to ready state
+     * 2. The operator checks if the rebalance resource is in `Ready` state or not
+     * 3. The Kafka cluster is now moved to `NotReady` state
+     * 4. The KafkaRebalance resource still remains in `Ready` state
```

Comment:
For step 1)  We cannot create a `KafkaRebalance` resource with updated status directly, we have to update the status after creation of the `KafkaRebalance`.  We are updating the status to `Ready` in step 1 itself(I have also mentioned in the documentation that we are moving it directly to ready state). For step 2) I guess since step 1) is moving the `KafkaRebalance` resource directly to ready state so we should check for sure that we are in `Ready` state or not and then do the part

Reply:
Thanks for the explanation @ShubhamRwt, makes sense to me!